Source/Particles/Resampling/ResamplingTrigger.H

```diff
@@ -0,0 +1,60 @@
+/* Copyright 2019-2020 Neil Zaim
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef WARPX_RESAMPLING_TRIGGER_H_
+#define WARPX_RESAMPLING_TRIGGER_H_
+
+#include "Utils/IntervalsParser.H"
+
+#include <AMReX_ParmParse.H>
+
+/**
+ * \brief This class is used to determine if resampling should be done at a given timestep for
+ * a given species. Specifically resampling is performed if the current timestep is included in
+ * the IntervalsParser m_resampling_intervals or if the average number of particles per cell of
+ * the considered species exceeds the threshold m_max_avg_ppc.
+ */
+class ResamplingTrigger
+{
+public:
+
+    /**
+     * \brief Constructor of the ResamplingTrigger class. Reads the resampling trigger parameters
+     * from the input file.
+     */
+    ResamplingTrigger ();
+
+    /**
+     * \brief A method that returns true if resampling should be done for the considered species
+     * at the considered timestep.
+     *
+     * @param[in] timestep the current timestep
+     * @param[in] global_numparts the total number of particles of the considered species
+     */
+    bool triggered (const int timestep, const amrex::Real global_numparts);
```

Comment:
Could this method be `const` ?
I know that in the first call it stores an internal parameter to be used in all the other calls, but since this does not
affect the result of the call, it can still be `const` ! However, you'll need to declare the affected internal variables as `mutable`. 
You may have a look here : https://arne-mertz.de/2017/10/mutable/

Reply:
Thanks for the link! From what's written inside, it indeed looks like a situation where it's good to use `mutable` so I think I'll do that.